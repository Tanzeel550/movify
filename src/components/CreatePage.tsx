import React from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { startCreateMovie } from '../actions/moviesAsyncActions';
import MovieForm from './FormComponent/MovieForm';
import VerifyEmailBtn from './Utils/VerifyEmailBtn';

type StateTypes = {
  auth: { user: { emailVerified: boolean } };
  history: { push: (link: string) => {} };
};

const mapStateToProps = ({ auth, history }: StateTypes) => ({
  emailVerified: auth.user.emailVerified,
  history,
});

const connector = connect(mapStateToProps, { startCreateMovie });
type PropsFromRedux = ConnectedProps<typeof connector>;

const CreatePage = (props: PropsFromRedux) => {
  return props.emailVerified ? (
    <MovieForm
      actionType="add"
      handleFormSubmit={async movie => {
        await props.startCreateMovie({ movie });
        props.history.push('/');
      }}
    />
  ) : (
    <div className="container jumbotron">
      <h3 className="u-text-colorized">
        You cannot access this page because your Email is not verified
      </h3>
      <VerifyEmailBtn />
    </div>
  );
};

export default connector(CreatePage);
