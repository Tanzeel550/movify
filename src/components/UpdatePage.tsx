import React from 'react';
import MovieForm from './FormComponent/MovieForm';
import { connect, ConnectedProps } from 'react-redux';
import {
  startDeleteMovie,
  startUpdateMovie,
} from '../actions/moviesAsyncActions';
import { Redirect } from 'react-router';
import { FireDBMovieItem } from '../types/APITypes';
import VerifiedComp from './Utils/VerifiedComp';

type StateTypes = {
  movies: FireDBMovieItem[];
  auth: { user: { emailVerified: boolean } };
};

type PropsType = {
  history: {
    push: (link: string) => {};
  };
  match: { params: { id: string } };
  movie: FireDBMovieItem;
};

const mapStateToProps = (state: StateTypes, props: PropsType) => ({
  movie: state.movies.find(movie => movie.id === props.match.params.id),
  emailVerified: state.auth.user.emailVerified,
  history: props.history,
});

const connector = connect(mapStateToProps, {
  startUpdateMovie,
  startDeleteMovie,
});

type Props = ConnectedProps<typeof connector>;

export const UpdatePage = ({
  movie,
  emailVerified,
  history,
  startDeleteMovie,
  startUpdateMovie,
}: Props) => (
  <VerifiedComp emailVerified={emailVerified}>
    {movie ? (
      <MovieForm
        actionType="update"
        movie={movie}
        handleFormSubmit={async data => {
          await startUpdateMovie({ id: movie!.id, movie: data });
          history.push('/');
        }}
        handleMovieDeletion={async () => {
          await startDeleteMovie({ id: movie!.id });
          history.push('/');
        }}
      />
    ) : (
      <h1 className="u-text-colorized">
        No Movie found on this page. Please go back and update some other Movie
        <button
          className="btn u-btn-colorized"
          onClick={() => <Redirect to="/" />}
        >
          Go Home
        </button>
      </h1>
    )}
  </VerifiedComp>
);

export default connector(UpdatePage);
