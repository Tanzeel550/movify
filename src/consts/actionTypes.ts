export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MoviesActionReturnType = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type MoviesActionParamsType = {
  id?: string;
  movie?: object;
  movies?: object;
};

export type SEARCH_TYPE = {
  text: string;
};

export type ErrorType = {
  message: string;
  title: string;
};

export type ErrorActionsReturnType = {
  type: 'SET_ERROR' | 'CLEAR_ERROR';
  message?: string;
  title?: string;
};
