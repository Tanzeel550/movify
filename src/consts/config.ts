const OMBD_API_KEY = process.env.REACT_APP_OMDB_API_KEY;
export const SEARCH_MOVIE_BY_TEXT = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&s=`;
export const GET_MOVIE_BY_TITLE = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&t=`;
export const GET_MOVIE_BY_IMDB_ID = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&i=`;

export const PROMISE_AREAS = {
  SEARCH_MOVIE_BY_TEXT: 'SEARCH_MOVIE_BY_TEXT',
  GET_MOVIE_BY_TITLE: 'GET_MOVIE_BY_TITLE',
  GET_ALL_MOVIES: 'GET_ALL_MOVIES',
  LOAD_PAGE: 'LOAD_PAGE',
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const TEST_MOVIES_SRC = 'movies_test';
