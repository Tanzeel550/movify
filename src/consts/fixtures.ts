import {
  APIMovieResultByTitle,
  APISingleMovieResultBySearch,
  FireDBMovieItem,
} from '../types/APITypes';
import { AuthUserType } from '../types/AuthTypes';

export const movies: FireDBMovieItem[] = [
  {
    id: '1234567',
    name: 'The Dark Knight',
    watched: true,
    dateWatched: '234567890',
    whatYouLearnt: 'The world is as good as we let it to be',
    Poster: 'https://image.com',
    createdAt: '9876543',
  },
  {
    id: '4567890',
    name: 'GodFather',
    watched: false,
    dateWatched: '98765434567',
    whatYouLearnt: 'You can act like a man what is the matter with you?',
    Poster: 'https://godfather.com',
    createdAt: '987654567890',
  },
  {
    id: '9ikjnbvfrty',
    name: 'Avengers: EndGame',
    watched: false,
    dateWatched: '',
    whatYouLearnt: 'Thanos will be defeated',
    Poster: 'https://avengers.com',
    createdAt: '69864123340404',
  },
];

export const myUser: AuthUserType = {
  user: {
    name: 'Tanzeel',
    email: 'tanzeelahmed550@gmail.com',
    passion: 'Web Dev',
    os: 'Kali Linux',
    version: '1.0.0',
    height: 5.5,
  },
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const email = 'unknownPerson123@uet.com';
export const password = 'JonSnow';
export const nameOfTheMovie = 'Godfather';

export const movieFromAPIByTitle: APIMovieResultByTitle = {
  Title: 'The Dark Knight',
  Year: '2008',
  Rated: 'PG-13',
  Released: '18 Jul 2008',
  Runtime: '152 min',
  Genre: 'Action, Crime, Drama, Thriller',
  Director: 'Christopher Nolan',
  Writer:
    'Jonathan Nolan (screenplay), Christopher Nolan (screenplay), Christopher Nolan (story), David S. Goyer (story), Bob Kane (characters)',
  Actors: 'Christian Bale, Heath Ledger, Aaron Eckhart, Michael Caine',
  Plot:
    'When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, Batman must accept one of the greatest psychological and physical tests of his ability to fight injustice.',
  Language: 'English, Mandarin',
  Country: 'USA, UK',
  Awards: 'Won 2 Oscars. Another 157 wins & 163 nominations.',
  Poster:
    'https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg',
  Ratings: [
    { Source: 'Internet Movie Database', Value: '9.0/10' },
    {
      Source: 'Rotten Tomatoes',
      Value: '94%',
    },
    { Source: 'Metacritic', Value: '84/100' },
  ],
  Metascore: '84',
  imdbRating: '9.0',
  imdbVotes: '2,341,244',
  imdbID: 'tt0468569',
  Type: 'movie',
  DVD: '14 Jun 2010',
  BoxOffice: '$534,858,444',
  Production: 'Syncopy',
  Website: 'N/A',
  Response: 'True',
};

export const moviesFromAPIBySearch: APISingleMovieResultBySearch[] = [
  {
    Title: 'The Dark Knight',
    Year: '2008',
    imdbID: 'tt0468569',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg',
  },
  {
    Title: 'The Dark Knight Rises',
    Year: '2012',
    imdbID: 'tt1345836',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTk4ODQzNDY3Ml5BMl5BanBnXkFtZTcwODA0NTM4Nw@@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns, Part 1',
    Year: '2012',
    imdbID: 'tt2313197',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMzIxMDkxNDM2M15BMl5BanBnXkFtZTcwMDA5ODY1OQ@@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns, Part 2',
    Year: '2013',
    imdbID: 'tt2166834',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BYTEzMmE0ZDYtYWNmYi00ZWM4LWJjOTUtYTE0ZmQyYWM3ZjA0XkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns',
    Year: '2013',
    imdbID: 'tt11060882',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMjExMGE4NDQtM2MwMS00YzhlLTlkYjgtNmExNzM1ZDBhYzg1XkEyXkFqcGdeQXVyMTY5Nzc4MDY@._V1_SX300.jpg',
  },
];
