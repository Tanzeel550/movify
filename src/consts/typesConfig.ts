export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MOVIES_ACTION = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type SEARCH_TYPE = {
  text: string;
};

export type MOVIE_ID = {
  id?: string;
  movie?: object;
  movies?: object;
};
