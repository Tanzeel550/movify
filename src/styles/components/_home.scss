.home__user--info {
  padding: 70px 0;
  background-color: $purple-color-1;
  margin-bottom: 100px;
  color: white;

  p {
    color: white;
  }
}
