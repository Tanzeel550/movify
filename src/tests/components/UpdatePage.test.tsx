import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { UpdatePage } from '../../components/UpdatePage';
import { movies } from '../../consts/fixtures';

let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>,
  startDeleteMovie: jest.Mock<any, any>,
  startUpdateMovie: jest.Mock<any, any>,
  history: { push: (link: string) => void };

beforeEach(() => {
  startUpdateMovie = jest.fn();
  startDeleteMovie = jest.fn();
  history = {
    push: (link: string) => {},
  };
  wrapper = shallow(
    <UpdatePage
      movie={movies[0]}
      emailVerified={true}
      startUpdateMovie={startUpdateMovie}
      startDeleteMovie={startDeleteMovie}
      // @ts-ignore
      history={history}
    />
  );
});

test('should render UpdatePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should not render UpdatePage because there is no movie found', () => {
  wrapper.setProps({ movie: null });
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('h1').text()).toMatch(
    'No Movie found on this page. Please go back and update some other Movie'
  );
});

export {};
