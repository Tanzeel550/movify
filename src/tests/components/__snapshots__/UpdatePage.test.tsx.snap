// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render UpdatePage because there is no movie found 1`] = `
<VerifiedComp
  emailVerified={true}
>
  <h1
    className="u-text-colorized"
  >
    No Movie found on this page. Please go back and update some other Movie
    <button
      className="btn u-btn-colorized"
      onClick={[Function]}
    >
      Go Home
    </button>
  </h1>
</VerifiedComp>
`;

exports[`should render UpdatePage correctly 1`] = `
<VerifiedComp
  emailVerified={true}
>
  <MovieForm
    actionType="update"
    handleFormSubmit={[Function]}
    handleMovieDeletion={[Function]}
    movie={
      Object {
        "Poster": "https://image.com",
        "createdAt": "9876543",
        "dateWatched": "234567890",
        "id": "1234567",
        "name": "The Dark Knight",
        "updatedAt": "",
        "watched": true,
        "whatYouLearnt": "The world is as good as we let it to be",
      }
    }
  />
</VerifiedComp>
`;
